airflow/contrib/operators/gcp_bigtable_operator.py

```diff
@@ -0,0 +1,401 @@
+        self.nodes = nodes
+        self._validate_inputs()
+        self.hook = BigTableHook()
+        super(BigTableClusterUpdateOperator, self).__init__(*args, **kwargs)
+
+    def execute(self, context):
+        instance = self.hook.get_instance(self.project_id, self.instance_id)
+        if not instance:
+            raise AirflowException("Instance '{}' does not exists.".format(self.instance_id))
+
+        try:
+            self.hook.update_cluster(
+                instance,
+                self.cluster_id,
+                self.nodes
+            )
+        except google.api_core.exceptions.NotFound:
+            raise AirflowException("Cluster '{}' does not exists for Instance '{}'.".format(
+                self.cluster_id,
+                self.instance_id
+            ))
+        except google.api_core.exceptions.GoogleAPICallError as e:
+            self.log.error('An error occurred. Exiting.')
+            raise e
+
+
+class BigTableTableWaitForReplicationSensor(BaseSensorOperator, BigTableValidationMixin):
+    """
+    Sensor that waits for BigTable Table to be fully replicated on its Clusters.
+
+    For more details about Cluster states for a Table have a look at the reference:
+    https://googleapis.github.io/google-cloud-python/latest/bigtable/table.html#google.cloud.bigtable.table.Table.get_cluster_states
+
+    :type project_id: str
+    :param project_id: The ID of the GCP Project
+    :type instance_id: str
+    :param instance_id: The ID of the Instance the Table belongs to
+    :type table_id: str
+    :param table_id: The ID of the Table to check replication status
+    """
+    REQUIRED_ATTRIBUTES = ('project_id', 'instance_id', 'table_id')
+    template_fields = ['project_id', 'instance_id', 'table_id']
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 instance_id,
+                 table_id,
+                 *args, **kwargs):
+        self.project_id = project_id
+        self.instance_id = instance_id
+        self.table_id = table_id
+        self._validate_inputs()
+        self.hook = BigTableHook()
+        super(BigTableTableWaitForReplicationSensor, self).__init__(*args, **kwargs)
+
+    def poke(self, context):
+        instance = self.hook.get_instance(self.project_id, self.instance_id)
+        if not instance:
+            self.log.info("Instance '%s' does not exists.", self.instance_id)
+            return False
+
+        try:
+            cluster_states = self.hook.get_cluster_states_for_table(instance, self.table_id)
+        except google.api_core.exceptions.NotFound:
+            self.log.info("Table '%s' does not exists in Instance '%s'.", self.table_id, self.instance_id)
+            return False
+
+        ready_state = ClusterState(enums.Table.ClusterState.ReplicationState.READY)
+        if not all([state == ready_state for state in cluster_states.values()]):
+            self.log.info("Table '%s' is not yet replicated.", self.table_id)
```

Comment:
Can we also log which cluster is not ready?

Reply:
sure